src/lxd.py

```diff
@@ -0,0 +1,388 @@
+
+        Raises:
+            LxdException: Unable to load the file to the LXD instance.
+        """
+        if isinstance(content, str):
+            content = content.encode("utf-8")
+
+        with tempfile.NamedTemporaryFile() as file:
+            file.write(content)
+            file.flush()
+
+            self.push_file(file.name, filepath, mode)
+
+    def pull_file(self, source: str, destination: str) -> None:
+        """Pull a file from the LXD instance.
+
+        Args:
+            source: Path of the file to pull in the LXD instance.
+            destination: Path of load the file.
+
+        Raises:
+            LxdException: Unable to load the file from the LXD instance.
+        """
+        lxc_cmd = [
+            "/snap/bin/lxc",
+            "file",
+            "pull",
+            f"{self.instance.name}/{source.lstrip('/')}",
+            destination,
+        ]
+
+        try:
+            execute_command(lxc_cmd)
+        except SubprocessError as err:
+            logger.exception("Failed to pull file")
+            raise LxdError(
+                f"Unable to pull file {source} from LXD instance {self.instance.name}"
+            ) from err
+
+    def read_file(self, filepath: str) -> str:
+        """Read content of a file in the LXD instance.
+
+        Args:
+            filepath: Path of the file in the LXD instance.
+
+        Raises:
+            LxdException: Unable to load the file from the LXD instance.
+
+        Returns:
+            The content of the file.
+        """
+        with tempfile.NamedTemporaryFile() as file:
+            self.pull_file(filepath, file.name)
+
+            return file.read().decode("utf-8")
+
+
+class LxdInstance:
+    """A LXD instance.
+
+    Attrs:
+        name (str): Name of LXD instance.
+        files (LxdInstanceFiles): Manager for the files on the LXD instance.
+        status (str): Status of the LXD instance.
+    """
+
+    def __init__(self, name: str, pylxd_instance: pylxd.models.Instance):
+        """Construct the LXD instance representation.
+
+        Args:
+            name: Name of the LXD instance.
+            pylxd_instance: Instance of pylxd.models.Instance for the LXD instance.
+        """
+        self.name = name
+        self._pylxd_instance = pylxd_instance
+        self.files = LxdInstanceFileManager(self._pylxd_instance)
+
+    @property
+    def status(self) -> str:
+        """Status of the LXD instance.
+
+        Returns:
+            Status of the LXD instance.
+        """
+        return self._pylxd_instance.status
+
+    def start(self, timeout: int = 30, force: bool = True, wait: bool = False) -> None:
+        """Start the LXD instance.
+
+        Args:
+            timeout: Timeout for starting the LXD instance.
+            force: Whether to force start the LXD instance.
+            wait: Whether to wait until the LXD instance started before returning.
+
+        Raises:
+            LxdException: Unable to start the LXD instance.
```

Comment:
LxdError?

Reply:
This was the old name. Should have update it.

Reply:
Minor comments